src/app/reservation/overview/overview.component.ts

```diff
@@ -150,38 +150,80 @@ export class OverviewComponent implements OnInit {
   }
 
   confirm() {
-    this.submitting = true;
 
+    if (!this.overviewForm.valid || this.selectedPaymentProvider == null) {
+      return; // prevent accidental submissions
+    }
+
+    this.submitting = true;
+    this.registerUnloadHook();
     this.selectedPaymentProvider.pay().subscribe(paymentResult => {
       if (paymentResult.success) {
-
         this.overviewForm.get('gatewayToken').setValue(paymentResult.gatewayToken);
-
         const overviewFormValue = this.overviewForm.value;
-
         this.reservationService.confirmOverview(this.eventShortName, this.reservationId, overviewFormValue, this.translate.currentLang).subscribe(res => {
           if (res.success) {
+            this.unregisterHook();
             if (res.value.redirect) { // handle the case of redirects (e.g. paypal, stripe)
               window.location.href = res.value.redirectUrl;
             } else {
               this.router.navigate(['event', this.eventShortName, 'reservation', this.reservationId, 'success']);
             }
           } else {
-            console.log('res is not success');
             this.submitting = false;
+            this.unregisterHook();
+            this.globalErrors = handleServerSideValidationError(res, this.overviewForm);
           }
         }, (err) => {
-          console.log('error');
           this.submitting = false;
+          this.unregisterHook();
           this.globalErrors = handleServerSideValidationError(err, this.overviewForm);
         });
       } else {
         console.log('paymentResult is not success (may be cancelled)');
+        this.unregisterHook();
         this.submitting = false;
       }
+    }, (err) => {
+      this.submitting = false;
+      this.unregisterHook();
+      this.notifyPaymentError(err);
     });
   }
 
+  private registerUnloadHook(): void {
+    console.log('warn on page reload: on');
+    // source: https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload
+    window.onbeforeunload = function (e: BeforeUnloadEvent) {
```

Comment:
use the syntax: `window.addEventListener('beforeunload', ...);` , so we don't override others beforeunload listeners